spellcheck/README.md

```diff
@@ -1,226 +1,182 @@
+Most of the existing metrics and evaluation algorithms compute the similarity between the reference and the prediction such as BLEU or ROUGE scores. Others calculate the Precision-Recall on modified tokens for token classification tasks.
 
-Model that wraps calls to the [RobotOFF API](https://github.com/openfoodfacts/robotoff). RobotOFF corrections are mainly based on some ElasticSearch suggestions. Rules are then applied to determine whether or not the suggestion is relevant.  
-The RobotOFF API has two parameters :
-- `index` :  can be either `product` (default), `product_all`, `product_extended` or `product_extended_all`]. Specify the vocabulary used by ES.
-- `confidence` : float, default to 1. Threshold parameter for ElasticSearch.
+But in our case, we would like to estimate how well the Spellcheck performs on recognizing and fixing the right elements in the list of ingredients. Therefore we need to compute the Precision-Recall of correctly modified tokens. 
 
+However, we don't have access to these tokens. Only to these text sequences:
+* The original: the list of ingredients to be corrected;
+* The reference: how we expect this list to be corrected;
+* The prediction: the correction from the model.
 
-At the moment, RobotOFF API is the model that handles almost all the spelling corrections. Especially, it performs well on misspelled words (maximum edit distance : 2). Calls are cached using joblib to speed up the tests so make sure to refresh the cache if RobotOFF behavior changed.
+Is there any way to get the Precision-Recall scores on corrected tokens only from these sequences? The answer is yes. This is the function of our evaluation algorithm: `scripts/benchmark/evaluation.py`.
 
-## Regex Model
+The uniqueness of this evaluation algorithm lies in its calculation of precision, recall, and F1 scores specifically for errors to be corrected, which are directly extracted from texts.
 
-Model containing several rule-based algorithms.
+The process is divided into 4 steps:
 
-### Pattern-based replacements
+1. Texts (Original-Reference-Prediction) are tokenized using a Byte Pair Encoding ([BPE](https://en.wikipedia.org/wiki/Byte_pair_encoding)) tokenizer from the [tiktoken](https://github.com/openai/tiktoken) library from OpenAI.
 
-Most basic algorithm that does some search and replace in the original string.  
-Patterns can be find in `./spellcheck/models/regex/patterns_fr.txt` . The file is divided in sections. Each section is composed of the correct form (first line) followed by common mistakes (following lines). The objective is to complete the document incrementally. Comments can be added using a `#` .  
-**Example :**
+Example:
+``` 
+Original:       "Th cat si on the fride,"
+Reference:      "The cat is on the fridge."
+Prediction:     "Th big cat is in the fridge."
 ```
-# This is a comment
 
-ingrédient
-ingredienb
-ingedients
-ingédients
+After tokenization:
+``` 
+Original:       791   8415   4502   389   79    282    1425   11
```

Comment:
The tokenization seems incorrect with the default tokenizer model, I have for original string:
`[1016, 8415, 4502, 389, 279, 282, 1425, 11]`